src/classes/TDTM_Config.cls

```diff
@@ -50,33 +50,65 @@ public class TDTM_Config {
     * @return List<TDTM_Global_API.TdtmToken> List of trigger handler tokens with the default values.
     *********************************************************************************************************/
     public static List<TDTM_Global_API.TdtmToken> getTdtmConfig() {
         if(tdtmTokensCache != null && tdtmTokensCache.size() > 0) {
             return tdtmTokensCache;
         }
-        List<TDTM_Global_API.TdtmToken> tokens = new List<TDTM_Global_API.TdtmToken>();
+
+        Boolean useAdv = false;
+        String apiHedaClass = '';
+        //Check if Advancement is installed and enabled.
+        try {
+            List<SObject> advSettings = Database.query('select Use_Advancement__c, API_HEDA_Class__c from Advancement_Setting__mdt');
+            useAdv = (Boolean)advSettings[0].get('Use_Advancement__c');
+            apiHedaClass = (String)advSettings[0].get('API_HEDA_Class__c');
+        } catch(Exception e) {
+            // Advancement not installed. Nothing to do. If the custom setting Adv_Settings doesn't exist, we'll use
+            // HEDA's TDTM config.
+        }
+
+        List<TDTM_Global_API.TdtmToken> tokens;
         List<Trigger_Handler__c> tdtmConfig = [select Class__c, Object__c, Trigger_Action__c, Load_Order__c, Active__c,
-                            Asynchronous__c, Filter_Field__c, Filter_Value__c, User_Managed__c, Owned_by_Namespace__c from Trigger_Handler__c];
+                            Asynchronous__c, Filter_Field__c, Filter_Value__c, User_Managed__c, Owned_by_Namespace__c
+                            from Trigger_Handler__c];
+
 
         // Getting the default configuration only if there is no data in the Trigger Handler object. Otherwise
         // we would delete customizations and Trigger Handlers entries that aren't in the default configuration.
-        if(tdtmConfig.size() == 0) {
-            tdtmConfig = TDTM_DefaultConfig.getDefaultRecords();
+        if(tdtmConfig == null || tdtmConfig.size() == 0) {
+            //No TDTM configuration in the database.
+            //If Advancement is installed and enabled, get the TDTM configuration from there.
+            if (useAdv) {
+                //Using Adv config.
+                Type classType = Type.forName(apiHedaClass);
+                if (classType != null) {
+                    Object classInstance = classType.newInstance();
+                    if (classInstance instanceof TDTM_Adapter) {
+                        TDTM_Adapter API_HEDA = (TDTM_Adapter) classInstance;
+                        //We already have the configuration as tokens, so no need for any transformation.
+                        tokens = API_HEDA.getDefaultTdtmConfigTokens();
+                    }
+                }
+            } else {
+                //Not Using Adv config.
+                tdtmConfig = TDTM_DefaultConfig.getDefaultRecords();
+            }
         }
 
+        tokens = new List<TDTM_Global_API.TdtmToken>();
```

Comment:
won't this overwrite the tokens you loaded from API_HEDA in line 94?

Reply:
True, need to fix this.